storage/store.go

```diff
@@ -1123,6 +1123,10 @@ func (s *Store) addRangeToRangeMap(rng *Range) error {
 func (s *Store) RemoveRange(rng *Range) error {
 	// RemoveGroup needs to access the storage, which in turn needs the
 	// lock. Some care is needed to avoid deadlocks.
+	if !s.stopper.StartTask() {
```

Comment:
This looks like the wrong place for a call to `StartTask`; we need to ensure that a task is started at a higher level before calling whatever calls RemoveRange. I assume this is via `EndTransaction -> mergeTrigger -> mergeRange`, in which case I think the right place to call StartTask is in `Store.processRaft` right after we've read an event from the multiraft channel. Once we're draining we should just drop multiraft events on the floor; they'll be retried by the next incarnation of the node.